src/query/sql/src/evaluator/block_operator.rs

```diff
@@ -88,201 +85,117 @@ impl BlockOperator {
+                                    // TODO(leiysky): this can be optimized by using a `zip` array function
+                                    row_result_set = row_result_set
+                                        .iter()
+                                        .map(|value| {
+                                            match value {
+                                                Value::<AnyType>::Scalar(_) => {
+                                                    // No need to pad scalar values.
+                                                    value.clone()
+                                                }
 
-    /// Unify all unnest columns. Make the length of each column be the same.
-    ///
-    /// Return the unified unnested columns and the unnest offsets for replicating non-unnest columns.
-    ///
-    /// If the original length of each column is different, the shorter ones will be padded with nulls.
-    ///
-    /// For example:
-    ///
-    /// ```
-    /// +---------+---------+
-    /// | origin1 | origin2 |
-    /// +---------+---------+
-    /// | [1,2]   | [1,2,3] |
-    /// | [3,4,5] | [4,5]   |
-    /// +---------+---------+
-    /// ```
-    ///
-    /// will be unnested to:
-    ///
-    /// ```
-    /// +---------+---------+
-    /// | unnest1 | unnest2 |
-    /// +------ --+---------+
-    /// | 1       | 1       |
-    /// | 2       | 2       |
-    /// | NULL    | 3       |
-    /// | 3       | 4       |
-    /// | 4       | 5       |
-    /// | 5       | NULL    |
-    /// +---------+---------+
-    /// ```
-    fn unify_unnest_columns(
-        unnest_columns: Vec<ArrayColumn<AnyType>>,
-    ) -> (Vec<Column>, Vec<usize>) {
-        debug_assert!(!unnest_columns.is_empty());
-        let num_rows = unnest_columns[0].len(); // Rows of the original `ArrayColumn`s.
-        debug_assert!(unnest_columns.iter().all(|col| col.len() == num_rows));
-        // Some pre-computed variables.
-        let unnest_columns = unnest_columns
-            .into_iter()
-            .map(|col| {
-                let typ = col.values.data_type().unnest();
-                let arrays = col.iter().map(|c| c.unnest()).collect::<Vec<_>>();
-                (typ, arrays)
-            })
-            .collect::<Vec<_>>();
+                                                Value::<AnyType>::Column(column) => {
+                                                    let nullable_column =
+                                                        column.as_nullable().unwrap().clone();
+                                                    let mut column_builder =
+                                                        NullableColumnBuilder::from_column(
+                                                            *nullable_column,
+                                                        );
+                                                    (0..(max_num_rows - result_set_rows)).for_each(
```

Comment:
Maybe operating the inner column and bitmap is more effective.

Reply:
I'll try it out.